app/views/components/button_component.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+class ButtonComponent < ApplicationComponent
+  include Phlex::Rails::Helpers::LinkTo
+  include Phlex::Rails::Helpers::ButtonTo
+
+  attr_reader :text, :href, :type, :options, :color_css, :size_css
+
+  def initialize(text = nil, href: nil, type: :button, color: :primary, outline: false, size: :large, **options) # rubocop:disable Metrics/ParameterLists
```

Comment:
This class seems somewhat complex, would it be worth putting a test together to ensure it works in various different scenarios?

Reply:
yes, I'll follow up with that.